File: classic puzzle-hard/the-hungry-duck---part-1.py
#https://www.codingame.com/training/hard/the-hungry-duck---part-1
import sys
import math

w,h=[int(i) for i in input().split()]
grid=[[0 for i in range(w+1)] for j in range(h+1)]
ans=[[0 for i in range(w+1)] for j in range(h+1)]

for i in range(h):
    arr=[int(j) for j in input().split()]
    for j in range(w):
        grid[i][j]=arr[j]

for i in range(1,h+1):
    for j in range(1,w+1):
        ans[i][j]=max(ans[i-1][j],ans[i][j-1])+grid[i-1][j-1]

print(ans[h][w])

File: classic puzzle-medium&above/the-hungry-duck---part-1.cpp
//https://www.codingame.com/training/hard/the-hungry-duck---part-1
#include <bits/stdc++.h>
using namespace std;
int grid[11][11];
int sum[11][11];

int main()
{
    int w;
    int h;
    cin>>w>>h; cin.ignore();
    for(int i=0;i<h;i++)
        for(int j=0;j<w;j++){
            int food;
            cin>>food; cin.ignore();
            grid[i][j]=food;
        }
    for(int i=1;i<=h;i++){
        for(int j=1;j<=w;j++){
            sum[i][j]=max(sum[i-1][j],sum[i][j-1])+grid[i-1][j-1];
        }
    }
    cout<<sum[h][w]<<endl;
}
